### ISSUE TITLE
404: Session not found 

### ISSUE THREAD

**Issue by Samadan01** [gradio-app/gradio#9252]
### Describe the bug

My response from the API during my GET request with command line interface(Bash), it keeps giving:
event: error
data: "404: Session not found."

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

curl -X POST https://samadan01-blip-image-captioning-api.hf.space/call/predict -s -H "Content-Type: application/json" -d '{
  "data": [
							{"path":"https://raw.githubusercontent.com/gradio-app/gradio/main/test/test_files/bus.png"}
]}' 

curl -N GET https://samadan01-blip-image-captioning-api.hf.space/call/predict/$EVENT_ID

### Logs

```shell
event: error
data: "404: Session not found."
```

### System Info

```shell
Using the lastest gradio on hugging Face, which runs perfectly on Hugging Face
```

### Severity

Blocking usage of gradio

**Comment by skye0402** [comment]
Yep, like #6920 and my bug #9169 - It's a nasty bug and my users complain. They can work around it by opening an incognito session in the browser until the regular browser works again (I think once a certain session timeout is reached which can be minutes or hours, I don't have details).

**Comment by AusafG5** [comment]
Is there any resolve for this ?

has almost made our prod apps unusable when deployed to googe cloud run services, localtunnelling, basically only works locally.

it's happening again in 5.7.0 version updates.

**Comment by yosun** [comment]
yes notice this error as well 

event: error
data: "404: Session not found."

**Comment by freddyaboulton** [comment]
Hi all! I can't repro this with this space: https://huggingface.co/spaces/gradio/image_mod_default_image

I just copy/pasted the snippet and it's working as expected. Is the EVENT_ID in the GET request matching the output from the POST request?

![Image](https://github.com/user-attachments/assets/ab7d72ee-f5b1-4b6f-ad8f-39a11095855f)

**Comment by AusafG5** [comment]
I don't know about anyone else but all of my projects which cause this have [Model Scope Waterfall Gallery](https://www.gradio.app/custom-components/gallery?id=modelscope%2Fmodelscope-studio). 
I can't really compare it by removing this, cuz the functionality which triggers this (like/dislike mechanism) is not in plain gradio.
